src/resolvers/project.ts

```diff
@@ -151,6 +204,85 @@ export const resolvers: Resolvers = {
         throw InternalServerError();
       }
     },
+
+    // Import project from an external data source
+    projectImport: async (_, {input}, context) => {
+      const reference = 'updateProject resolver';
+      try {
+        if (isAuthorized(context.token)) {
+          try {
+            const projectId = input.project.id;
+            const existingProject = await Project.findById(reference, context, projectId);
+            if (!existingProject) {
+              throw NotFoundError();
+            }
+
+            if (!hasPermissionOnProject(context, existingProject)) {
+              throw ForbiddenError();
+            }
+
+            // Update project
+            const toUpdateProject = new Project(input.project);
+            const updatedProject = await toUpdateProject.update(context);
+            if(!updatedProject){
+              toUpdateProject.addError('general', 'Unable to update project');
+              return toUpdateProject;
+            }
+
```

Comment:
It's possible for the `toUpdateProject.update` function to return the project with errors (instead of just null). It's probably good to check if it has errors , something like this: 
```
if (updatedProject && !updatedProject.hasErrors()) {
  // ... add the funders
  // ... add the contributors 
}
```